From b2b88ae3385e209c2120a2611900860d8e9f8706 Mon Sep 17 00:00:00 2001
From: Eric Lundby <Eric.Lundby@gmail.com>
Date: Thu, 22 Jan 2026 11:12:40 -0700
Subject: [PATCH] Apply the patches to the TensorFlow repository during Bazel build.

Apply the patches created in 0005 and 0006 to the TensorFlow repository during Bazel build. 
---
 WORKSPACE | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/WORKSPACE b/WORKSPACE
index 8100e6a..e6473e2 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -77,6 +77,8 @@ http_archive(
         "//third_party/tensorflow:tensorflow.grappler_optimizers.BUILD.patch",
         "//third_party/tensorflow:tensorflow.xla_allocator_registry.BUILD.patch",
         "//third_party/tensorflow:tensorflow.core.ops.BUILD.patch",
+        "//third_party/tensorflow:abseil_systemlib.patch",
+        "//third_party/tensorflow:protobuf_systemlib.patch",
     ],
     sha256 = "0e1494185594ce7a6c1cf86c984da2f2d9bef24427104cb2c76c1ccee76ee986",
     strip_prefix = "tensorflow-2.21.0",
@@ -163,6 +165,27 @@ load("//third_party/tensorflow:tf_configure.bzl", "tf_configure")
 
 tf_configure()
 
+new_local_repository(
+    name = "pypi_tensorflow",
+    path = "__SYSTEM_TF_PATH__",
+    build_file_content = """
+cc_library(
+    name = "tf_header_lib",
+    hdrs = glob(["include/**/*"]),
+    strip_include_prefix = "include/",
+    visibility = ["//visibility:public"],
+)
+cc_library(
+    name = "libtensorflow_framework",
+    srcs = select({
+        "@bazel_tools//src/conditions:darwin": ["libtensorflow_framework.2.dylib"],
+        "//conditions:default": ["libtensorflow_framework.so.2"],
+    }),
+    visibility = ["//visibility:public"],
+)
+""",
+)
+
 load("@pypi//:requirements.bzl", "install_deps")
 
 install_deps()
